Sorting/Bucket_Sort/BucketSort.py

```diff
@@ -66,4 +63,16 @@ def bucketSort(numbers):
     return sortedNumbers
 
 # testing Code
-print(bucketSort(numbers))
\ No newline at end of file
+# For purpose of this code, enter numbers only between (0.0 and 1.0)
+
+if __name__ == "__main__":
+
+## numbers is our array, 
+## Assumption about our array - 
+# All values are floating point values between 0.0 and 1.0.
+# This helps our bucketing scheme. For different data, we will have different schemes
+    numbers = input("Enters numbers >= 0 and  < 1 for sorting: ")
```

Comment:
Is it a must that the numbers should be between 0 and 1?

Reply:
This is non comparison based sorting algorithm. So, I think we need to make some assumptions about our data. 

Since in this code we are using `numOfBuckets = 10`, and our bucketing function is `int(element * numOfBuckets)` so we can do this sort with floating point values in [0, 1) range for this case. 

If we use a value let's say 1.1, then that will give us the value of 11, which will be out of index.

Reply:
Okay. Can it be done, that the inputs, which are not between 0 and 1, are normalized to values between 0 and 1, by dividing with a specific divisor. The distribution will remain the same. After performing bucket sort, the resultant values are multiplied again with the divisor before, giving the original values. In python specifically I think this can be done pretty easily using numpy. Can you do this?

Reply:
Okay. Yes that can definitely be done. In that line I came up with an idea. 

Let us while assigning buckets, take the sigmoid of the values.`sigmoid(x) = 1 / 1+exp(-x)` This will put the values in the range `[0,1]` and deal with all kinds of values.

Then we will assign buckets as:
```python
##  sigmoid defined above
numBuckets = 11   #### [0, 1, 2..., 9, 10]
for element in numbers:
    ## All buckets is dictionary containing the buckets
    allBuckets[sigmoid(element) * (numBuckets - 1)].append(element)
    #numbuckets - 1 as otherwise we can get index 11
```
Since sigmoid will be used only while assigning buckets, so after assigning buckets, we proceed normally as before.

How does this idea sound? Should I proceed with it?

Reply:
How will you get the original values after applying the sigmoid function?

Reply:
For the **purpose of assigning buckets only** we will use the the sigmoid function. We will be storing the **original values only** in the dictionary, not the sigmoid converted versions

Reply:
Okay, great idea. Go ahead!

Reply:
Okay I have added the functionality. Do review it.